src/python-kolibri-flatpak-utils/kolibri_redirect.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python3
+
+# Displays a loading screen, and then redirects to Kolibri once it is
+# responding. Stops automatically if it does not receive a request after a
+# certain amount of time.
+
+import http.server
+import socketserver
+import threading
+import time
+
+from urllib.parse import urlparse, parse_qs
+
+from .utils import KOLIBRI_URL, get_is_kolibri_responding
+
+
+WWW_DIR = '/app/www'
+
+KOLIBRI_REDIRECT_IDLE_TIMEOUT_SECS = 60
```

Comment:
Hmm, those variable names are pretty awkward. Feel free to change them if you think of a clearer scheme :) Indeed, the thought is the redirect service may as well exit on a short timer, since it's a one-off thing on a random port, and each time the launcher gets started a different redirect service is spawned (so if one stops getting requests, it won't expect another client to connect).